### ISSUE TITLE
Referencing a secret key

### ISSUE THREAD

**Issue by Pyttei** [bitnami/charts#11296]
### Name and Version

bitnami/logstash 5.0.12

### What steps will reproduce the bug?

How do I reference a secret key? I have added a secret key to Logstash values.yaml using 
```yaml
extraEnvVarsSecret: my-secret
```

The only key, value pair in the secret is:
```yaml
my-secret-password: [my-password]
```
Then I want to send data to Elasticsearch and this secret includes my Elasticsearch password so I have tested:
```
password => "${my-secret-password}"
```
But this doesn't work. If I hard code the password it works as expected.

### What is the expected behavior?

I expect connection to elasticsearch to work.

### What do you see instead?

```
[WARN ][logstash.outputs.elasticsearch][main] Attempted to resurrect connection to dead ES instance, but got an error {:url=>"https://elastic:xxxxxx@elasticsearch:9200/", :exception=>LogStash::Outputs::ElasticSearch::HttpClient::Pool::BadResponseCodeError, :message=>"Got response code '401' contacting Elasticsearch at URL 'https://elasticsearch:9200/'"}
```

**Comment by Pyttei** [comment]
@alemorcuq did you have any answer for this?

**Comment by alemorcuq** [comment]
According to [Logstash's documentation](https://www.elastic.co/guide/en/logstash/current/environment-variables.html), that's how you should reference environment variables.

I suspect the issue is with your environment variable name, `my-secret-password`. If I run this from a bash console:

```
I have no name!@ss-logstash-0:/opt/bitnami/logstash$ echo ${my-secret-password}
secret-password
```

Can you try using a different environment variable name (i.e. a different key in the secret)? For example, `MY_SECRET_PASSWORD`. And then use it like this from your Logstash configuration: `password => "${MY_SECRET_PASSWORD}"`

**Comment by Pyttei** [comment]
Hi @alemorcuq ,
They key comes from bitnami elasticsearch. If you set `security.enabled: true` in elasticsearch values.yaml but leave everything else as default a Secret called `elasticsearch`  is created, then the only key: value pair is `elasticsearch-password: [elastic-password]`. In the kibana config I can reference this key by just setting `kibana.elasticsearch.security.auth.elasticsearchPasswordSecret: elasticsearch `

So basically I want logstash config to reference this key so I don't have to hardcode it into logstash. Is there some similar way to do this in the Logstash config as in the elasticsearch bundled kibana config? Or is there some other way?

**Comment by alemorcuq** [comment]
Oh, I see the issue now. The secret is created automatically so you don't have control over the key name. May you can add an environment variable like this?

```
- name: ES_PASSWORD
        valueFrom:
          secretKeyRef:
            name: my-secret
            key: elasticsearch-password
```

And then reference it in Logstash's configuration as:

```
password => "${ES_PASSWORD}"
```

**Comment by Pyttei** [comment]
Thank you @alemorcuq ,
This worked good. I think this could be good info to add to Logstash README since many people want to use full elastic stack. Or it could also be a nice possibility if an option `elasticsearchPasswordSecret` was added to Logstash config.
